### PR TITLE
Cloud code coverage

### PR DESCRIPTION
**Pull request by spencer-skylight**
# Description
* Abstract base class methods have comment to exclude them from code coverage.
* Add test case to validate AzureCredentialManager _need_new_token() returns true before token has been assigned
* Tweak test cases for GCP credential manager to test default scope, and `project_id` getter
* Test edge case in FHIR export download for downloading exports with no exported content.

### CODE REVIEW

File: tests/cloud/test_cloud.py [hunk 1]
```diff
@@ -209,11 +214,14 @@ def test_gcp_credential_manager_handle_expired_token(mock_gcp_creds, mock_gcp_re
...
 
-    credential_manager = GcpCredentialManager(scope=scope)
+    credential_manager = GcpCredentialManager()
 
     # When invalid credentials are encountered they should be refreshed by the
     # credential manager when a new token is requested.
-    _ = credential_manager.get_credential_object()
+    assert credential_manager.get_project_id() is not None
+    scoped_credentials = credential_manager.get_credential_object()
+    scoped_credentials == credential_manager.scoped_credentials
```
Comment (KennethSkylight) [inline tests/cloud/test_cloud.py:223]: I think this line isn't doing anything since you're doing a comparison without a conditional like `if`

File: phdi/cloud/core.py [hunk 2]
```diff
@@ -13,14 +13,14 @@ def get_credential_object(self) -> object:
         """
         Returns a cloud-specific credential object
         """
-        pass
+        pass  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (spencer-skylight) [reply phdi/cloud/core.py:16]: That tells coverage.py to exclude the line since we don't have a great way (or reason) to have our unit tests execute it.  Here's the documentation for that:
https://coverage.readthedocs.io/en/6.3.3/excluding.html

File: tests/cloud/test_cloud.py (hunk 1 above)
Comment (spencer-skylight) [reply tests/cloud/test_cloud.py]: Nice catch. Added `assert` for the second line of code.

File: phdi/cloud/core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bamader) [reply phdi/cloud/core.py:16]: @KennethSkylight This will also be clarified after I merge in my geospatial tests PR, where I researched practices around this stuff and documented it all in our repo with our other testing information.

**Review by bamader — APPROVED** [review]
Great work!

**Review by KennethSkylight — APPROVED** [review]
LGTM
